src/Components/Web.JS/src/Platform/WebView/WebViewIpcSender.ts

```diff
@@ -22,7 +22,8 @@ export function sendEndInvokeJSFromDotNet(asyncHandle: number, succeeded: boolea
 }
 
 export function sendByteArray(id: number, data: Uint8Array) {
-  send('ReceiveByteArrayFromJS', id, data);
+  const dataBase64Encoded = btoa(String.fromCharCode.apply(null, data as unknown as number[]));
```

Comment:
[Some stackoverflowers are worried](https://stackoverflow.com/questions/12710001/how-to-convert-uint8-array-to-base64-encoded-string) that this technique can fail if the data contains multibyte unicode characters. I don't know if their concerns are valid, but since our goal here is to transport arbitrary developer-supplied data, we should allow for all possible sequences of bytes.

Do you know whether or not the concerns about particular byte sequences are valid?

Reply:
I don't believe this is a major concern for us at this time. The technique of `Uint8Array` -> `Base64 string` used here is the same as what we [used previously when reading file data](https://github.com/dotnet/aspnetcore/pull/33015/files#diff-64e870656d425c992a8f001f7bc462e19ae9816d663b6d93e8cb0e6d1b04fea4R104-R107).

Additionally, there doesn't seem to be a clear standardized/library solution, and I'm a bit hesitant to integrate some of the scripts proposed. For instance, the `TextDecoder` approach may have browser compatibility issues, and requires us to assume `utf-8`. This may resolve the multi-byte unicode char issue, however it may break other use cases. Further, due to the complementary stream interop you've proposed, large byte array conversion shouldn't be a factor here.

I think this may be one of those things we can revisit should users run into this issue. If you feel differently, please let me know and I can address accordingly.